pgpy/pgp.py

```diff
@@ -250,9 +252,18 @@ def revocable(self):
 
     @property
     def revocation_key(self):
-        if 'RevocationKey' in self._signature.subpackets:
-            raise NotImplementedError()
-        return None
+        """
+        A ``list`` of revocation key subpackets in this signature, if any. Otherwise, an empty ``list``.
+
+        Each is a namedtuple with the following attributes:
+        ``revocation_key.keyclass`` - a ``list`` of :py:obj:`~pgpy.constants.RevocationKeyClass` flags.
+
+        ``revocation_key.algorithm`` - the :py:obj:`~pgpy.constants.PubkeyAlgorithm` of the revocation key.
+
+        ``revocation_key.fingerprint`` - the :py:obj:`~pgpy.types.Fingerprint` of the revocation key.
+        """
+        return list(self._revocation_key(rk.keyclass, rk.algorithm, rk.fingerprint)
+                    for rk in self._signature.subpackets['RevocationKey'])
```

Comment:
so one thing that we discussed on IRC is that this should really be a `set`, not a list: there's no reason why we'd want duplicate items here. @Commod0re also suggested to return only fingerprints here, but i can imagine there is some value in extracting the other data from here. what's the use of `keyclass` and `algorithm` anyways?

Reply:
Well AFAIK RFC4880 says that both fingerprint, algorithm and keyclass have to match for the key to be considered a valid revocation key, so thats why I returned it all so a user can filter keys based on all of those properties.

Reply:
okay, makes sense.